src/nvim/drawscreen.c

```diff
@@ -699,20 +699,32 @@ void end_search_hl(void)
   screen_search_hl.rm.regprog = NULL;
 }
 
-static void win_border_redr_title(win_T *wp, ScreenGrid *grid, int col)
+static void win_redr_bordertext(win_T *wp, ScreenGrid *grid, VirtText text_chunks, int row, int col)
 {
-  VirtText title_chunks = wp->w_float_config.title_chunks;
-
-  for (size_t i = 0; i < title_chunks.size; i++) {
-    char *text = title_chunks.items[i].text;
+  for (size_t i = 0; i < text_chunks.size; i++) {
+    char *text = text_chunks.items[i].text;
     int cell = (int)mb_string2cells(text);
-    int hl_id = title_chunks.items[i].hl_id;
+    int hl_id = text_chunks.items[i].hl_id;
     int attr = hl_id ? syn_id2attr(hl_id) : 0;
-    grid_puts(grid, text, 0, col, attr);
+    grid_puts(grid, text, row, col, attr);
     col += cell;
   }
 }
 
+int win_border_get_text_col(int total_col, int text_width, AlignTextPos align)
```

Comment:
maybe `win_get_bordertext_col` as "bordertext" seems to be used regularly for either header/footer.

Reply:
I opted for not using `bordertext` as it doesn't depend on `bordertext_type`.

Reply:
But it still is only used for these two things. I don't think there is anything more it could be used for as we now have filled both the top and bottom border parts with (potential) text, right?

Reply:
> But it still is only used for these two things. I don't think there is anything more it could be used for as we now have filled both the top and bottom border parts with (potential) text, right?

I had a crazy idea of maybe allowing text on left and right parts, but can't see useful use case.